lib/ukofw/fdt.c

```diff
@@ -391,3 +392,39 @@ bool fdt_prop_read_bool(const void *fdt, int start_offset,
 
 	return prop ? true : false;
 }
+
+int fdt_chosen_stdout_path(const void *fdt, int *offs, char **opt)
+{
+	const char *prop;
+	int prop_len, len;
+	int nchosen;
+	char *p;
+
+	UK_ASSERT(fdt);
+	UK_ASSERT(offs);
+	UK_ASSERT(opt);
+
+	nchosen = fdt_path_offset((void *)fdt, "/chosen");
+	if (unlikely(nchosen < 0)) {
+		uk_pr_err("Could not find 'chosen' node (%d)\n", nchosen);
+		return -FDT_ERR_NOTFOUND;
+	}
+
+	prop = fdt_getprop((void *)fdt, nchosen, "stdout-path",
+			   &prop_len);
+	if (unlikely(!prop || prop_len <= 0)) {
+		uk_pr_warn("Could not find 'stdout-path' property\n");
+		return -FDT_ERR_NOTFOUND;
+	}
+
+	p = strchr(prop, ':');
+
+	*opt = p ? p + 1 : NULL;
```

Comment:
This is just a nitpick and most likely just my personal taste. But I feel like
```C
if (p) {
   *opt = p + 1;
   len = p - prop;
} else {
   *opt = NULL;
   len = prop_len - 1;
}
```
would be easier to follow and maybe more pleasant during a GDB debugging session if someone happens to have to go through here. This also makes it easier to add individual prints for the two branches so that one may be able to visually debug logs, I believe.

Reply:
I think it's a matter of personal taste, and as a matter of fact I find the terse notation cleaner.

Reply:
Ok. Just thought that someone may easily lose sight of the fact that `len` depends on the state of `p` after the `*opt` statement, instead of its state before that (I personally did).